red_october/models/red_october_user.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+import logging
+
+from odoo import _, api, fields, models
+from odoo.http import request
+from odoo.exceptions import ValidationError
+
+
+_logger = logging.getLogger(__name__)
+
+
+class RedOctoberUser(models.Model):
+
+    _name = 'red.october.user'
+    _description ='Red October User'
+
+    SESSION_USER_ATTR = 'ro_uid'
+
+    user_id = fields.Many2one(
+        string='User',
+        comodel_name='res.users',
+        required=True,
+    )
+    name = fields.Char(
+        related='user_id.name',
+    )
+    vault_ids = fields.Many2many(
+        string='Vaults',
+        comodel_name='red.october.vault',
+        required=True,
+        default=lambda s: s._default_vault_ids(),
+        domain="[('user_id.company_id', 'in', company_ids)]",
+    )
+    delegation_ids = fields.One2many(
+        string='Delegations',
+        comodel_name='red.october.delegation',
+        inverse_name='user_id',
+    )
+    is_admin = fields.Boolean(
+        string='Admin',
+    )
+    is_active = fields.Boolean()
+    encrypt_type = fields.Selection([
+        ('RSA', 'RSA'),
+        ('ECC', 'ECC'),
+    ],
+        default='RSA',
+        required=True,
+    )
+
+    @api.model
+    def _default_vault_ids(self):
+        return [
+            (6, 0, self.env['red.october.vault'].get_current_vault().ids),
+        ]
+
+    @api.multi
+    @api.constrains('user_id', 'vault_ids')
+    def _check_user_id_vault_ids(self):
+        for record in self:
+            results = self.search([
+                ('user_id', '=', record.user_id.id),
+                ('vault_ids', 'in', record.vault_ids.ids),
+            ])
+            if len(results) > 1:
```

Comment:
This is a constraint, so we're testing for more than one. I think we're good in this instance?